src/trials.jl

```diff
@@ -289,6 +290,32 @@ function withtypename(f, io, t)
     end
 end
 
+function bindata(sorteddata, nbins)
+    min, max = sorteddata[[1; end]]
+    Δ = (max - min) / nbins
+    bins = zeros(nbins)
+    lastpos = 0
+    for i ∈ 1:nbins
+        pos = searchsortedlast(sorteddata, min + i * Δ)
+        bins[i] = pos - lastpos
+        lastpos = pos
+    end
+    bins
+end
+
+function asciihist(bins, height=1)
+    histbars = ['▁', '▂', '▃', '▄', '▅', '▆', '▇', '█']
+    if minimum(bins) == 0
+        barheights = 2 .+ round.(Int, (height * length(histbars) - 2) * bins ./ maximum(bins))
+        barheights[bins .== 0] .= 1
+    else
+        barheights = 1 .+ round.(Int, (height * length(histbars) - 1) * bins ./ maximum(bins))
+    end
+    [let bheight = barheights[b] - (h-1) * length(histbars)
```

Comment:
I find this form hard to read, can we use a `map` or explicit for loop instead?

Reply:
Refactored to:
```jl
    heightmatrix = [min(length(histbars), barheights[b] - (h-1) * length(histbars))
                    for h in height:-1:1, b in 1:length(bins)]
    map(height -> if height < 1 ' ' else histbars[height] end, heightmatrix)
```